src/components/__tests__/QueryRuleCustomData.js

```diff
@@ -30,6 +32,41 @@ it('renders in a list of <pre> by default', () => {
 `);
 });
 
+it('gives the items to the main slot', () => {
+  const items = [{ text: 'this is user data' }, { text: 'this too!' }];
+  __setState({
+    items,
+  });
+
+  mount(QueryRuleCustomData, {
+    scopedSlots: {
+      default(props) {
+        expect(props).toEqual({
+          items,
+        });
+      },
+    },
+  });
+});
+
+it('gives individual items to the item slot', () => {
+  const items = [{ text: 'this is user data' }, { text: 'this too!' }];
+  expect.assertions(items.length);
+  __setState({
+    items,
+  });
+
+  mount(QueryRuleCustomData, {
+    scopedSlots: {
+      item(props) {
+        expect(props).toEqual({
+          item: expect.objectContaining({ text: expect.any(String) }),
```

Comment:
I suppose this could have been done with a spy:

```js
const itemScope = jest.fn();

mount(QueryRuleCustomData, {
  scopedSlots: {
    item: itemScope,
  },
});

expect(itemScope).toHaveBeenCalledWith({ item: items[0] });
expect(itemScope).toHaveBeenCalledWith({ item: items[1] });
```